clang/lib/CIR/Lowering/ThroughMLIR/LowerCIRToMLIR.cpp

```diff
@@ -468,30 +468,27 @@ class CIRSinOpLowering : public mlir::OpConversionPattern<cir::SinOp> {
 };
 
 template <typename CIROp, typename MLIROp>
-class CIRBitOpLowering : public mlir::OpConversionPattern<CIROp> {
+class CIRCountZerosBitOpLowering : public mlir::OpConversionPattern<CIROp> {
 public:
   using mlir::OpConversionPattern<CIROp>::OpConversionPattern;
 
   mlir::LogicalResult
   matchAndRewrite(CIROp op,
                   typename mlir::OpConversionPattern<CIROp>::OpAdaptor adaptor,
                   mlir::ConversionPatternRewriter &rewriter) const override {
-    auto resultIntTy = mlir::cast<mlir::IntegerType>(
-        this->getTypeConverter()->convertType(op.getType()));
-    auto res = rewriter.create<MLIROp>(op->getLoc(), adaptor.getInput());
-    auto newOp = createIntCast(rewriter, res->getResult(0), resultIntTy,
-                               /*isSigned=*/false);
-    rewriter.replaceOp(op, newOp);
+    if (op.getIsZeroPoison()) {
+      return op.emitError() << "Unable to lower potentially poisoned "
+                            << op.getOperationName() << " to math dialect";
+    }
```

Comment:
Well I believe we can reasonably ignore this `zero_poison` flag here. `zero_poison` basically means the computation result is undefined, and it's valid to behave like zero is not poisonsed.